src/Components/Debugger.fs

```diff
@@ -195,6 +198,187 @@ module Debugger =
+            | _ -> ()
+
+            if JS.isDefined ls.environmentVariables then
+                let vars =
+                    ls.environmentVariables.Keys
+                    |> Array.map (fun k -> k, box ls.environmentVariables[k])
+
+                c?env <- createObj vars
+
+
+            c?console <- "internalConsole"
+            c?stopAtEntry <- false
+            let presentation = {| hidden = false; group = "ionide" |}
+
+            c?presentation <- presentation
+
+            Some c
+
+    let configsForProject (project, launchSettings: JsMap<LaunchSettingsConfiguration>) =
+        seq {
+            for name in launchSettings.Keys do
+                logger.Info $"Making config for {name}"
+                let settings: LaunchSettingsConfiguration = launchSettings[name]
+
+                if JS.isDefined settings then
+                    match makeDebugConfigFor (name, settings, project) with
+                    | Some cfg -> yield cfg
+                    | None -> ()
+                else
+                    ()
+        }
+
+    let defaultConfigForProject (p: Project) : DebugConfiguration option =
+        if p.OutputType <> "exe" then
+            None
+        else
+            let c = createEmpty<DebugConfiguration>
+            c.name <- $"{path.basename p.Project}"
+            c.``type`` <- "coreclr"
+            c.request <- "launch"
+            c?program <- p.Output
+
+            c?cwd <- "${workspaceFolder}"
+
+            c?console <- "internalConsole"
+            c?stopAtEntry <- false
+            let presentation = {| hidden = false; group = "ionide" |}
+
+            c?presentation <- presentation
+            Some c
+
+    let launchSettingProvider =
+        { new DebugConfigurationProvider with
+            override x.provideDebugConfigurations(folder: option<WorkspaceFolder>, token: option<CancellationToken>) =
+                logger.Info $"Evaluating launch settings configurations for workspace '%A{folder}'"
+
+                let configs =
+                    Project.getInWorkspace ()
+                    |> Seq.choose (function
+                        | Project.ProjectLoadingState.Loaded x -> Some x
+                        | x ->
+                            logger.Info $"Discarding project '{x}' because it is not loaded"
+                            None)
+                    |> Seq.collect (fun (p: Project) ->
+                        match readSettingsForProject p with
+                        | Some launchSettings ->
+                            let projectConfigs = configsForProject (p, launchSettings)
+
+                            if Seq.isEmpty projectConfigs then
+                                (Option.toList (defaultConfigForProject p)
+                                 |> List.toSeq)
+                            else
+                                projectConfigs
```

Comment:
Should we also append the default one in case the user have `launchSettings` ?

Reply:
My thinking was something like: "having a launchSettings file means a user made an explicit choice in how the app needs to be run, so we should defer to that".

That might not always be the case, and/or the file might be used for certain specific launch _modes_, so I can see the argument for adding the default configuration too. I mostly worry about choice overload for the user, but a) we limit this to runnable project types already, b) the list is filterable by typing, and c) recently-used configurations are pinned to the top of the list, so I think that's a good UX overall.